pkg/kubelet/dockertools/labels.go

```diff
@@ -108,3 +146,78 @@ func getIntValueFromLabel(labels map[string]string, label string) (int, error) {
 	// Just set the value to 0
 	return 0, nil
 }
+
+func getInt64PointerFromLabel(labels map[string]string, label string) (*int64, error) {
+	if strValue, found := labels[label]; found {
+		int64Value, err := strconv.ParseInt(strValue, 10, 64)
+		if err != nil {
+			return nil, err
+		}
+		return &int64Value, nil
+	}
+	// Because it's normal that a container has no PodDeletionGracePeriod and PodTerminationGracePeriod label,
+	// don't report any error here.
+	return nil, nil
+}
+
+// getJsonObjectFromLabel returns a bool value indicating whether an object is found
+func getJsonObjectFromLabel(labels map[string]string, label string, value interface{}) (bool, error) {
+	if strValue, found := labels[label]; found {
+		err := json.Unmarshal([]byte(strValue), value)
+		return found, err
+	}
+	// Because it's normal that a container has no PreStopHandler label, don't report any error here.
+	return false, nil
+}
+
+// The label kubernetesPodLabel is added a long time ago (#7421), it serialized the whole api.Pod to a docker label.
+// We want to remove this label because it serialized too much useless information. However kubelet may still work
+// with old containers which only have this label for a long time until we completely deprecate the old label.
+// Before that to ensure correctness we have to supply information with the old labels when newly added labels
+// are not available.
+// TODO(random-liu): Remove this function when we can completely deprecate kubernetesPodLabel
+func supplyContainerInfoWithOldLabel(labels map[string]string, containerInfo *labelledContainerInfo) {
+	// Get api.Pod from old label
+	var pod *api.Pod
+	data, found := labels[kubernetesPodLabel]
+	if !found {
+		// Don't report any error here, because it's normal that a container has no pod label, especially
+		// when we gradually deprecate the old label
+		return
+	}
+	pod = &api.Pod{}
+	err := latest.GroupOrDie(api.GroupName).Codec.DecodeInto([]byte(data), pod)
+	if err != nil {
+		// If the pod label can't be parsed, we should report an error
+		logError(containerInfo, kubernetesPodLabel, err)
+		return
+	}
+	// The pod could never be nil here.
```

Comment:
Don't think the comment is necessary since you already did the assignment in line 188